### ISSUE TITLE
The effect of torch.manual_seed() on a simple graph classifier. 

### ISSUE THREAD

**Issue by CodePanda9** [pyg-team/pytorch_geometric#2049]
Running a simple GCN for classifying graphs.  There are around 6000 graphs, all labeled. I want to run multiple times, and compute a confidence interval for the accuracy. I used the `torch.manual_seed()` function for this, but I give it a random seed every time: 

```
class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        **torch.manual_seed(np.random.randint(100, size=1)[0])**
        self.conv1 = GCNConv(dataset_num_features, 16)
        self.lin = nn.Linear(16, dataset_num_classes)

    def forward(self, data):
        x, edge_index, batch = data.x, data.edge_index, data.batch
        x = self.conv1(x, edge_index)
        x = x.relu()
        x = global_mean_pool(x, batch)  
        x = F.dropout(x, p=0.8, training=self.training)
        x = self.lin(x)
        return x
```

I realized that, if I run this network 10 times, I only get two values for the test accuracy: 

```
Training run : 1
Test accuracy : 0.8278481012658228
Training run : 2
Test accuracy : 0.8278481012658228
Training run : 3
Test accuracy : 0.17215189873417722
Training run : 4
Test accuracy : 0.17215189873417722
Training run : 5
Test accuracy : 0.8278481012658228
Training run : 6
Test accuracy : 0.8278481012658228
Training run : 7
Test accuracy : 0.17215189873417722
Training run : 8
Test accuracy : 0.17215189873417722
Training run : 9
Test accuracy : 0.8278481012658228
Training run : 10
Test accuracy : 0.17215189873417722
```

Why is there no variation besides these 2 numbers? In general, why if I run the network without a seed, I get `0.8278481012658228` every single time?

**Comment by rusty1s** [comment]
That's hard to tell from solely looking at test accuracies. In general, if you do not set a seed, weight initializations are different each time you re-run the script. You can confirm that by looking at the weights of `nn.Linear.weight` or by looking at the loss values during training (those should be definitely different).
